Buildings/Controls/OBC/ASHRAE/PrimarySystem/BoilerPlant/Staging/SetPoints/SetpointController.mo

```diff
@@ -0,0 +1,596 @@
+        origin={80,-80})));
+
+  Buildings.Controls.OBC.CDL.Interfaces.BooleanInput uBoiAva[nBoi]
+    "Boiler availability status vector"
+    annotation (Placement(transformation(extent={{-440,-190},{-400,-150}}),
+      iconTransformation(extent={{-140,-180},{-100,-140}})));
+
+  Buildings.Controls.OBC.CDL.Interfaces.BooleanInput uPla
+    "Plant enable signal"
+    annotation (Placement(transformation(extent={{-440,-100},{-400,-60}}),
+      iconTransformation(extent={{-140,-120},{-100,-80}})));
+
+  Buildings.Controls.OBC.CDL.Interfaces.IntegerInput u(
+    final min=0,
+    final max=nSta)
+    "Boiler stage"
+    annotation (Placement(transformation(extent={{-440,-130},{-400,-90}}),
+      iconTransformation(extent={{-140,-150},{-100,-110}})));
+
+  Buildings.Controls.OBC.CDL.Interfaces.RealInput THotWatRetPri(
+    final unit="K",
+    final displayUnit="K",
+    final quantity="ThermodynamicTemperature") if not primaryOnly
+    "Measured temperature of return hot water in primary circuit"
+    annotation (Placement(transformation(extent={{-440,30},{-400,70}}),
+      iconTransformation(extent={{-140,-30},{-100,10}})));
+
+  Buildings.Controls.OBC.CDL.Interfaces.RealInput THotWatRetSec(
+    final unit="K",
+    final displayUnit="K",
+    final quantity="ThermodynamicTemperature") if not primaryOnly
+    "Measured temperature of return hot water in secondary circuit"
+    annotation (Placement(transformation(extent={{-440,-10},{-400,30}}),
+      iconTransformation(extent={{-140,-60},{-100,-20}})));
+
+  Buildings.Controls.OBC.CDL.Interfaces.RealInput THotWatSupSet(
+    final unit="K",
+    final displayUnit="K",
+    final quantity="ThermodynamicTemperature")
+    "Hot water supply temperature setpoint"
+    annotation (Placement(transformation(extent={{-440,270},{-400,310}}),
+      iconTransformation(extent={{-140,150},{-100,190}})));
+
+  Buildings.Controls.OBC.CDL.Interfaces.RealInput THotWatSup(
+    final unit="K",
+    final displayUnit="K",
+    final quantity="ThermodynamicTemperature")
+    "Measured hot water supply temperature"
+    annotation (Placement(transformation(extent={{-440,150},{-400,190}}),
+      iconTransformation(extent={{-140,60},{-100,100}})));
+
+  Buildings.Controls.OBC.CDL.Interfaces.RealInput THotWatRet(
+    final unit="K",
+    final displayUnit="K",
+    final quantity="ThermodynamicTemperature")
+    "Hot water return temperature"
+    annotation (Placement(transformation(extent={{-440,230},{-400,270}}),
+      iconTransformation(extent={{-140,120},{-100,160}})));
+
+  Buildings.Controls.OBC.CDL.Interfaces.RealInput VHotWat_flow(
+    final quantity="VolumeFlowRate",
+    final unit="m3/s",
+    final displayUnit="m3/s")
+    "Measured hot water flow rate"
+    annotation (Placement(transformation(extent={{-440,190},{-400,230}}),
+      iconTransformation(extent={{-140,90},{-100,130}})));
+
+  Buildings.Controls.OBC.CDL.Interfaces.RealInput VMinSet_flow[nSta](
+    final unit="m3/s",
```

Comment:
Just as a reminder please apply the `fill` function consistently for any vector variable.